examples/DrawXNative/DrawXShared/RealmDraw.cs

```diff
@@ -0,0 +1,572 @@
+                    _currentColorIndex = SwatchColor.Colors.IndexOf(_currentColor);
+                }
+            }
+        }
+        #endregion Settings
+
+        public RealmDraw(float inWidth, float inHeight)
+        {
+            // TODO close the Realm
+            _canvasWidth = inWidth;
+            _canvasHeight = inHeight;
+
+            // simple local open            
+            // _realm = Realm.GetInstance("DrawX.realm");
+
+            _pencilBitmaps = new List<SKBitmap>(SwatchColor.Colors.Count);
+            foreach (var swatch in SwatchColor.Colors)
+            {
+                _pencilBitmaps.Add(EmbeddedMedia.BitmapNamed(swatch.Name + ".png"));
+            }
+
+            _loginIconBitmap = EmbeddedMedia.BitmapNamed("CloudIcon.png");
+        }
+
+        internal void InvalidateCachedPaths()
+        {
+            _redrawPathsAtNextDraw = true;
+            _hasSavedBitmap = false;
+            _currentlyDrawing = null;
+        }
+
+        internal async void LoginToServerAsync()
+        {
+            // in case have lingering subscriptions, clear by clearing the results to which we subscribe
+            _allPaths = null;
+
+            _waitingForLogin = true;
+            var s = Settings;
+            //// TODO allow entering Create User flag on credentials to pass in here instead of false
+            var credentials = Credentials.UsernamePassword(s.Username, s.Password, false);
+            var user = await User.LoginAsync(credentials, new Uri($"http://{s.ServerIP}"));
+            Debug.WriteLine($"Got user logged in with refresh token {user.RefreshToken}");
+
+            var loginConf = new SyncConfiguration(user, new Uri($"realm://{s.ServerIP}/~/Draw"));
+            _realm = Realm.GetInstance(loginConf);
+            #if DEBUG
+            _realm.RealmChanged += (sender, e) => 
+            {
+                Debug.WriteLine("Extra Realm Update callback invoked");
+            };
+            #endif
+            _allPaths = _realm.All<DrawPath>() as RealmResults<DrawPath>;
+            _allPaths.SubscribeForNotifications((sender, changes, error) =>
+            {
+                // WARNING ChangeSet indices are only valid inside this callback
+                if (changes == null)  // initial call
+                {
+                    RefreshOnRealmUpdate();  // force initial draw on login
+                    return;
+                }
+                //// we assume if at least one path deleted, drastic stuff happened, probably erase all
+                if (_allPaths.Count() == 0 || changes.DeletedIndices.Length > 0)
+                {
+                    Debug.WriteLine($"Realm notifier: Invalidating on Realm change as paths deleted or none");
+                    InvalidateCachedPaths();  // someone erased their tablet
+                    RefreshOnRealmUpdate();
+                    return;
+                }
+
+                int numInserted = changes.InsertedIndices.Length;
```

Comment:
I don't think we enforce it, just feel that we should try to keep our examples clean and consistent and we mostly use `var`s, that's why I pointed it out. It's insignificant though, so feel free to ignore it if you don't feel like changing it :p